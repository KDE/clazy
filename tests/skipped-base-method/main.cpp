#include <QtCore/QString>

class Base {
public:
    virtual void method1() {}
    virtual void method2() {}
    virtual void method3() {}
};

class Derived : public Base {
public:
    virtual void method1() override {}
    void method3() override = 0;
};

class DerivedDerived : public Derived {
public:
    void test()
    {
        Base::method1(); // Warn
        Derived::method1(); // OK
        method1();

        Base::method2(); // OK
        Base::method3(); // OK
    }
};
