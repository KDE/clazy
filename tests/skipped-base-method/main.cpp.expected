skipped-base-method/main.cpp:20:9: warning: Maybe you meant to call Derived::method1() instead [-Wclazy-skipped-base-method]
